# Frequent-k-mers-in-SARS-CoV-2

In this study, we have implemented FrequentWords() and BetterFrequentWords() algorithms to find frequent k-mers which works on counting and hashing principle, respectively. It has been found hashing by BetterFrequentWords() performs tasks much faster as compared to Frequentwords(). To test our best algorithm, we considered multiple virus’s genomes – SARS-CoV-2, SARS-CoV, MERS, Bat rat and Pangolin. In SARS-CoV-2 genome, there’s poly (A) tail at the end of its genome sequence which was removed to make our results more biologically convincing. We have reported multiple length k-mer sequences in SARS-CoV-2 and some of them also found in other viral genomes. Functional annotation of genes is also reported in which candidate k-mer sequences were found from Uniprot database. No motifs were found for selected k-mer sequences using RCSB PDB motif finder.
